### ISSUE TITLE
Unable to log in. (Udemy for business custom company domain)

### ISSUE THREAD

**Issue by GabrielCoetzee** [FaisalUmair/udemy-downloader-gui#126]
Hi,

I am completely unable to log in. It is probably due to the account being Udemy for business, and on a company domain. "companyx.udemy.com". Is there perhaps a workaround?

I keep getting either "Invalid Username/Password" or "Connection Failiure".

Yes, I am on the latest version. (1.6)

**Comment by FaisalUmair** [comment]
Hello @GabrielCoetzee What do you type in `Udemy Business Name` field?

It should only be the name not entire domain name. For example for `apple.udemy.com`, you have to type `apple` only.

**Comment by GabrielCoetzee** [comment]
Hi @FaisalUmair ,

I am only typing in the "company" part in "company,udemy.com" as you indicated.

**Comment by FaisalUmair** [comment]
Few days ago, I tried creating a trial business account for testing the app against it.

But it appeared to work fine, I don't know if a real working business account is different from that. But I guess it should work for real business account if it works for trial account.

And I also found that Udemy creates two separate account for you, one associated with Udemy business and other with normal Udemy website. Did you try skipping the Udmey business field? In order to check if the normal account works.

**Comment by GabrielCoetzee** [comment]
I did try that too. I`ve also tried a variable of lowercase and uppercase characters in both my username and 'Udemy Business' field in case it is case-sensitive, but also no success.

I think something must be different with my Udemy For Business account then. It is company provided, so I can`t use my credentials to log into normal "www.udemy.com". I have to log in through my browser at "company_name_omitted.udemy.com". It only works there.

Should I maybe try also creating a 'normal' udemy account, using the same credentials as my 'Udemy For Business' account, and give that a shot? Maybe both are required for it to work?

**Comment by GabrielCoetzee** [comment]
I`ve also tried both the "x86" and "x64" versions of Udeler, as well as running in elevated mode vs not running in elevated mode. Doesn`t seem to have an effect,

**Comment by FaisalUmair** [comment]
Yeah, just try to create a normal account. If the normal account doesn't work for you. Then the login is not working for you at all.

Since you mentioned that sometimes you get Connection Failed error, it makes me think it could be connection related issue. Are you using a proxy server?

**Comment by GabrielCoetzee** [comment]
I can`t create a normal account since the email address is already in use.

No, I`m not using a proxy server.

**Comment by FaisalUmair** [comment]
Just create a account with another email address, in order to check if the login works for you using normal account.

**Comment by donjordano** [comment]
I have same issue. I can log in with my normal account, but not with my company. Previously worked very well, I even managed to download some courses. Also when I'm with my normal account in the last version 1.6 it not listing the courses I have in my account.

**Comment by GabrielCoetzee** [comment]
Hi @FaisalUmair - I created a normal udemy account, and I can successfully log in. However, my 'Udemy for Business' account obviously still persists.

Hi @donjordano - Can you please elaborate? Are you saying uninstall the old version and do a clean install of the new version? (I`m on Windows 10)

**Comment by FaisalUmair** [comment]
Well, I am trying with my business account and it appears to work fine. If I enter correct username and password but incorrect business name, it still logs in but shows no course. But when I login with the same credentials and type correct business name it shows me courses from my business account. If I uncheck/skip Udemy business, it just login in to normal account. 

The login function since v1.5.4 has changed. So that could interfere with business account. It could also be possible that your business account and the trial business account I created are different. 

Now in order for me to fix this, I would have to reproduce this issue on my side. If anyone here can share their credentials (through email), which work on Udemy website but not on Udeler, I might be able to resolve this issue.

**Comment by GabrielCoetzee** [comment]
Hi @FaisalUmair, 

Yeah there must be something different with my account vs your trial business account then. There is clearly no normal udemy account linked to mine so no login works for me, business or otherwise. Whether I enter company name, no company name, uncheck "Udemy Business?", I`ve never had a successful login. 

I completely understand that in order to fix it, you need to reproduce the issue. However, my udemy for business account is provided by the company I work for and providing you with the credentials would be the highest possible breach of security. I understand this means we are at a dead-end, but it`s fine. I`ll have to download the videos one at a time with a chrome extension. :)

**Comment by FaisalUmair** [comment]
@GabrielCoetzee I understand the issue with sharing credentials. Unfortunately for issues related to Udemy account and specific paid courses, I would obviously need to have access to reproduce. 

I will see what else I can do about this.

**Comment by jcortejoso** [comment]
So as far as I could debug I have enabled the devtools in the app. It seems the authentication endpoint in the app is POST `https://www.udemy.com/api-2.0/auth/udemy-auth/login/?fields[user]=access_token`, but in my business page is POST to `https://<business>.udemy.com/join/login-popup/`.

**Comment by FaisalUmair** [comment]
@jcortejoso  I received your email. Thanks for providing you credentials to debug.

I will explain the difference in endpoints, `https://www.udemy.com/api-2.0/auth/udemy-auth/login/?fields[user]=access_token` is used to authenticate no matter what your account type is. 

Now for retrieving the course if we use `https://www.udemy.com/api-2.0/users/me/subscribed-courses?page_size=1` it will try to get courses from normal Udemy account, but since your account is a business account you need to request `https://sngular.udemy.com/api-2.0/users/me/subscribed-courses?page_size=1`. This functionality is already implemented in the app.

I was able to login and view courses from you business account. if you check the is_business checkbox and enter `sngular` in the field you will be able to view your courses. However if you leave it blank or don't check is_business at all, you will still be able to login but it wont show you any courses as your normal account doesn't have any courses.

Let me know if that is also the case with you.

And others please let me know what error you are getting and make sure you are entering correct business name in the field. 

I understand that the business login in the app is quite deceiving sometimes you think you were able to login because you already typed correct details. But as I mentioned before you will still be able to login even if you enter incorrect business name but correct credentials.

**Comment by manu1222** [comment]
I'm getting the same issue. Incorrect username/ password. But actually, it's correct.

**Comment by akhilpadgilwar** [comment]
> Hi @FaisalUmair,
> 
> Yeah there must be something different with my account vs your trial business account then. There is clearly no normal udemy account linked to mine so no login works for me, business or otherwise. Whether I enter company name, no company name, uncheck "Udemy Business?", I`ve never had a successful login.
> 
> I completely understand that in order to fix it, you need to reproduce the issue. However, my udemy for business account is provided by the company I work for and providing you with the credentials would be the highest possible breach of security. I understand this means we are at a dead-end, but it`s fine. I`ll have to download the videos one at a time with a chrome extension. :)

Hi Gabriel,

I am facing the exact same issue.
Did you find any other way to download videos from UDEMY? 
or are you using chrome extension only?
let me know if you got any solution for this.

Thanks,
Akhil

**Comment by iamguilhermelima** [comment]
Same issue here

**Comment by akhilpadgilwar** [comment]
He has not solved this issue?

**Comment by FaisalUmair** [comment]
Hi @akhilpadgilwar No one supplied me with their login credentials to debug the issue. I requested it multiple times. I got one email recently asking if I need the credentials, I replied to it, but got no response.  

And your business account seems to use a different login technique than other accounts. (I am not sure about this), but if you can also send me the credentials, I might be able to get some clue.

I will mention this again, I tried to create a business account and it worked for me. So whoever is facing this issue and want the issue to be fixed, kindly email me your login credentials so that I can debug the issue.

And I am not able to work on the issues raised here, due to my busy schedule :( I hope to get some free time soon, to work on a new update and fix majority of issues.

**Comment by rhamallac** [comment]
Hi @theWebArtisans!

what is supposed to be written at Business box? just "myCompany" or "mycompany.udemy.com"?

Thanks in advance!

**Comment by karlochacon** [comment]
hope you get back to us with good news for udemy business accounts

**Comment by purbarunc** [comment]
When I am going to log in with the email for the udemy business it is opening another window and when I returned to the previous screen it again opened another window. This continued over and over again. Hence I couldn`t just type the email.

**Comment by FnZnLi** [comment]
Ok go to udemy right click or F12 for **Inspect**
Choose **Application** tab, then choose **Cookies**
Take **access_token** form there
https://user-images.githubusercontent.com/50474221/57762361-e6eb8700-76ee-11e9-954f-e75dfe0cab5d.png
Whşle loggin in udeler use **Access Token** and **paste it**
https://i.imgur.com/nITcv04.jpg

**Comment by KingBrute** [comment]
> So as far as I could debug I have enabled the devtools in the app. It seems the authentication endpoint in the app is POST `https://www.udemy.com/api-2.0/auth/udemy-auth/login/?fields[user]=access_token`, but in my business page is POST to `https://<business>.udemy.com/join/login-popup/`.

Hello dude can u say how u ssl unpinn udemy app plz
